example/prometheus/main.go

```diff
@@ -67,23 +72,33 @@ func initMeter() {
 func main() {
 	initMeter()
 
-	meter := global.Meter("ex.com/basic")
+	// TODO Bring back Global package
+	// meter := global.Meter("ex.com/basic")
+	meter := meterProvider.Meter("ex.com/basic")
 	observerLock := new(sync.RWMutex)
 	observerValueToReport := new(float64)
 	observerLabelsToReport := new([]attribute.KeyValue)
-	cb := func(_ context.Context, result metric.Float64ObserverResult) {
+
+	gaugeObserver, err := meter.AsyncFloat64().Gauge("ex.com.one")
+	if err != nil {
+		log.Panicf("failed to initialize instrument: %v", err)
+	}
+	_ = meter.RegisterCallback([]instrument.Asynchronous{gaugeObserver}, func(ctx context.Context) {
 		(*observerLock).RLock()
 		value := *observerValueToReport
 		labels := *observerLabelsToReport
 		(*observerLock).RUnlock()
-		result.Observe(value, labels...)
-	}
-	_ = metric.Must(meter).NewFloat64GaugeObserver("ex.com.one", cb,
-		metric.WithDescription("A GaugeObserver set to 1.0"),
-	)
+		gaugeObserver.Observe(ctx, value, labels...)
+	})
 
-	histogram := metric.Must(meter).NewFloat64Histogram("ex.com.two")
-	counter := metric.Must(meter).NewFloat64Counter("ex.com.three")
+	histogram, err := meter.SyncFloat64().Histogram("ex.com.two")
```

Comment:
I do really miss Must() for instrument initialization

Reply:
I agree that the Must() forms tend to make the code read better, but they really don't make sense in this world. Both because this would either mean extra API surface, but also because our meter <-> instrument relation is different from both Prometheus and opencensus.  Both had a static implementation of our meter, and it could create new instruments at init.  That model doesn't work very well when you have to initialize your meter before you register instruments.